internal/api/core/user_server.go

```diff
@@ -54,3 +56,67 @@ func MustNewServerUserInfo(user *model.ServerUser) *ServerUserInfo {
 
 	return info
 }
+
+func NewServerUserInfos(users []*model.ServerUser) ([]*ServerUserInfo, error) {
+	result := make([]*ServerUserInfo, 0, len(users))
+	for _, user := range users {
+		info, err := NewServerUserInfo(user)
+		if err != nil {
+			return nil, fmt.Errorf("Failed to get server user info for user (%s).", user.Email)
+		}
+
+		result = append(result, info)
+	}
+
+	slices.SortFunc(result, CompareServerUserInfoPointer)
+
+	return result, nil
+}
+
+func CompareServerUserInfoPointer(a *ServerUserInfo, b *ServerUserInfo) int {
+	if a == b {
+		return 0
+	}
+
+	if a == nil {
+		return 1
+	}
+
+	if b == nil {
+		return -1
+	}
+
+	return strings.Compare(a.Email, b.Email)
+}
+
+// Get server user info from a generic map (like what an API response would have).
+func ServerUserInfoFromMap(data map[string]any) *ServerUserInfo {
```

Comment:
Do we need this?
Do we also need `CourseUserInfoFromMap`?
(Normally you don't touch things not related to your PR, but I'll give special permission for this.)